packages/data-point/lib/reducer-types/reducer-object/factory.js

```diff
@@ -26,42 +32,61 @@ function isType (source) {
 
 module.exports.isType = isType
 
+/**
+ * @return {Object}
+ */
+function newProps () {
+  return {
+    constants: {},
+    reducers: []
+  }
+}
+
+module.exports.newProps = newProps
+
 /**
  * @param {Function} createReducer
  * @param {Object} source
- * @param {Array} path
- * @param {Array} props
+ * @param {Array} stack
+ * @param {Object} props
  * @returns {Array}
  */
-function getReducerProps (createReducer, source, path = [], props = []) {
+function getProps (createReducer, source, stack = [], props = newProps()) {
   for (let key of Object.keys(source)) {
+    const path = stack.concat(key)
     const value = source[key]
-    const fullPath = path.concat(key)
     if (isPlainObject(value)) {
       // NOTE: recursive call
-      getReducerProps(createReducer, value, fullPath, props)
+      getProps(createReducer, value, path, props)
       continue
     }
 
-    props.push({
-      path: fullPath,
-      reducer: createReducer(value)
-    })
+    const reducer = createReducer(value)
+    if (reducer.type === 'ReducerConstant') {
```

Comment:
i guess my assumtions were wrong, so help me understand, if the constant is declared as the only reducer for a property then its set as static here, but it can also be used as a constant reducer anywhere else?

Reply:
Yes, the constant reducer can be used anywhere, like this for example:

```js
// constant inside a ReducerList (just for the sake of example)
const reducer = [constant({ a: 1 }), { a: '$a' }]
dataPoint.transform(reducer, {})
```

That `constant` will use this resolve function, but constants inside of ReducerObjects are just added to the static object so they never use this resolve.